src/wallet/wallet.cpp

```diff
@@ -2820,6 +2818,12 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CWalletT
                     }
                 }
 
+                // If subtracting fee from recipients, we now know what fee we
+                // need to subtract, we have no reason to reselect inputs
+                if (nSubtractFeeFromAmount && nFeeRet == 0) {
```

Comment:
Actually, because of
```cpp
                    // Never create dust outputs; if we would, just
                    // add the dust to the fee.
                    if (IsDust(newTxOut, discard_rate))
```
`nFeeRet` can be `> 0`, and in that case there is no need to pick new inputs too.